python/openchange/provision.py

```diff
@@ -920,6 +1096,50 @@ def openchangedb_deprovision(names, lp, mapistore=None):
     openchangedb.remove()
 
 
+# migration methods for openchange directory
+def directory_migrate(lp, uri=None, version=None, extra=None):
+    provisioned = False
+    try:
+        check_not_provisioned(extra['names'], lp, extra['creds'])
+    except:
+        provisioned = True
+    if not provisioned:
+        raise Exception("OpenChange is not provisioned, provision it instead of upgrading it")
+
```

Comment:
In other migrations, the first migration is in charge of provision the system. You should use the same approach here and having the 1st migration as the provision one.